docs/how-to-guides/deploy.md

```diff
@@ -0,0 +1,279 @@
+and [Ansible](https://www.ansible.com/) installed. It is recommended to install
+`ansible` using `pip`:
+
+```
+pip install ansible
+```
+
+Once done, make sure that you're in the `infrastructure/terraform` directory,
+and run the following two commands to provision the VM for SoftPack:
+
+```
+terraform init
+terraform apply
+```
+
+…following the onscreen instructions to confirm the actions that will take
+place.
+
+## Installing the Software
+
+While still in the `infrastructure/terraform` directory, you can run the
+following command to set-up the newly provisioned VMs with the necessary
+software and configuration:
+
+```
+ansible-playbook --inventory ansible/hosts.yml ansible/main.yml
+```
+
+You will need to response 'yes' to each of the three SSH authenticity checks,
+but should otherwise not be required to interact with the process.
+
+NB: This will most likely take a few minutes.
+
+## After Installation Modification
+
+Once the initial software configuration is done, you'll need to log in to the
+machine to make some changes:
+
+```
+ssh ubuntu@builder.softpack-USERNAME.hgi-dev.sanger.ac.uk
+```
+
+Once logged in, you'll need to create a directory and modify some files.
+
+Create the needed directory with the following command:
+
+```
+mkdir /opt/softpack/gpg/
+```
+
+The first file to edit is in the softpack builder configuration file, located at
+`/opt/softpack/builder/softpack_builder/config/conf/config.yml`. In the `vault`
+section, You'll need to fill in the details for the URL, Path, and Token for the
+vault, either with real values, or dummy ones such as the following:
+
+```
+vault:
+  url: https://something.com
+  path: /something
+  token: something
+```
+
+For the file `~/.softpack/builder/config.yml`, in the
+`spack->manifest->spack->container->images` section, remove the following three
+lines:
+
+```
+        images:
+          os: ubuntu:20.04
+          spack: v0.20.0
+```
+
+In the file `/lib/systemd/system/prefect-agent.service`, you'll need to append
+the spack path to the `Environment` line, like so:
+
+```
+Environment=PATH=/opt/softpack/builder/.venv/bin:/usr/local/bin:/usr/bin:/opt/spack/v0.20.1/bin
+```
+
+Lastly, in the `/opt/softpack/builder/softpack_builder/singularity.py` file,
+find and delete the following line:
+
+```
+				file.writelines(commands)
+```
+
+## Restart the SoftPack Services
+
+With the changes made, you should now be able to restart both the Prefect Agent
+and SoftPack Builder services, with the following commands:
+
+```
+sudo systemctl daemon-reload
+sudo systemctl restart prefect-agent
+sudo systemctl restart softpack-builder
+```
+
+## Building Software
+
+You should now be able to build software.
+
+First, you need to activate the correct environment, which you can do with the
+following command:
+
+```
+cd /opt/softpack/builder/ && source .venv/bin/activate
+```
+
+Once run, you should be able to install software with the following command:
```

Comment:
It would be good to have some "development/testing" instructions/explanation in here as well. For example, I'm too frightened to actually run the softpack-builder command because I don't know what it's going to do. Will it write to our artifacts repo? Could it screw something important up? Or is it only doing things on my OpenStack VM, where it's harmless?